packages/dataparcels/src/parcel/Parcel.js

```diff
@@ -187,23 +187,23 @@ export default class Parcel {
         }
     };
 
-    _changeAndReturn = (changeCatcher: (parcel: Parcel) => void): Parcel => {
-        let changedParcel = this;
-        let {_onHandleChange} = this;
+    _changeAndReturn = (changeCatcher: (parcel: Parcel) => void): ?[Parcel, ChangeRequest] => {
+        let result;
+        let {_onHandleChange, _lastOriginId} = this;
 
         // swap out the parcels real _onHandleChange with a spy
-        this._onHandleChange = (parcel) => {
-            changedParcel = parcel;
-            changedParcel._onHandleChange = _onHandleChange;
+        this._onHandleChange = (parcel, changeRequest) => {
+            // _changeAndReturn should not alter _lastOriginId
+            // as it's never triggered by a user action
+            // so revert to the current parcel's _lastOriginId
+            parcel._onHandleChange = _onHandleChange;
+            parcel._lastOriginId = _lastOriginId;
+            result = [parcel, changeRequest];
         };
 
         changeCatcher(this);
         this._onHandleChange = _onHandleChange;
-        // _changeAndReturn should not alter _lastOriginId
-        // as it's never triggered by a user action
-        // so revert to the current parcel's _lastOriginId
-        changedParcel._lastOriginId = this._lastOriginId;
-        return changedParcel;
+        return result;
```

Comment:
Is this safe? The result is nullable so in cases where you are restructuring off it could cause errors.

Reply:
In theory its unsafe, but all usages are internal and the case of not having a tuple in the result isnt possible given the usages. But good catch, I've added in a fix and tests so that if this function ever switches to be part of the public api then its already safe